todoman/cli.py

```diff
@@ -224,49 +247,49 @@ def edit(ctx, id, todo_properties, interactive):
             setattr(todo, key, value)
 
     if interactive or (not changes and interactive is None):
-        ui = TodoEditor(todo, ctx.obj['db'].lists(), ctx.obj['formatter'])
+        ui = TodoEditor(todo, ctx.db.lists(), ctx.ui_formatter)
         state = ui.edit()
         if state == EditState.saved:
             changes = True
 
     if changes:
         todo.save()
         if old_list.name != todo.list.name:
-            database.move(todo, todo.list, from_list=old_list)
-        click.echo(ctx.obj['formatter'].detailed(todo))
+            ctx.db.move(todo, todo.list, from_list=old_list)
+        click.echo(ctx.formatter.detailed(todo))
     else:
         click.echo('No changes.')
-        ctx.exit(1)
+        sys.exit(1)
```

Comment:
`ctx` is now our `AppContext`, not `click.Context`. I didn't find it very apealing to pass the click ctx just because of that.